src/metadataGeneration/customAttribute.ts

```diff
@@ -0,0 +1,67 @@
+import * as ts from 'typescript';
+import { getInitializerValue } from './initializer-value';
+import { Tsoa } from './tsoa';
+
+const getArgumentValue = (argument: ts.Expression) => {
+  switch (argument.kind as ts.SyntaxKind) {
+    case ts.SyntaxKind.StringLiteral: {
+      const argumentValue = (argument as any).text;
+
+      return argumentValue;
+    }
+
+    case ts.SyntaxKind.ObjectLiteralExpression: {
+      const argumentValueProperties = (argument as any).properties;
+
+      const argumentValue = {};
+
+      for (const property of argumentValueProperties) {
+        const name = property.name.text;
+        const scopes = getInitializerValue(property.initializer);
+        argumentValue[name] = scopes;
+      }
+
+      return argumentValue;
+    }
+
+    case ts.SyntaxKind.ArrayLiteralExpression: {
+      const elements = (argument as any).elements;
+
+      const argumentValue = elements.map((element) => getArgumentValue(element));
+
+      return argumentValue;
+    }
+
+    default: {
+      throw new Error('Custom Attribute values must be of type: string, object, or array');
+    }
+  }
+};
+
+export function getCustomAttributes(decorators: ts.Identifier[]): Tsoa.CustomAttribute[] {
+  const customAttributes: Tsoa.CustomAttribute[] = decorators.map((customAttributeDecorator: ts.Identifier) => {
+    const expression = customAttributeDecorator.parent as ts.CallExpression;
+
+    const [decoratorKeyArg, decoratorValueArg] = expression.arguments;
+
+    if (decoratorKeyArg.kind !== ts.SyntaxKind.StringLiteral) {
+      throw new Error('First argument of Custom Attribute must be the attribute key as a string');
+    }
+
+    const attributeKey = getArgumentValue(decoratorKeyArg);
```

Comment:
I think it would be easier to read is this would just check for 
```ts
if (!ts.isStringLiteral(decoratorKeyArg)) {
  throw new Error('The first argument of @CustomAttribute must be the attribute key as a string');
}
```
and subsequently just accessed `decoratorKeyArg.text`.
The call to getArgumentValue is unnecessary for the `decoratorKeyArg`.